### ISSUE TITLE
Agent.multiaction_probabilities is not batched in "marginal" sampling mode

### ISSUE THREAD

**Issue by Arthur031221** [infer-actively/pymdp#444]
`Agent.multiaction_probabilities` maps over the batch axis in `"full"` mode and not in `"marginal"` mode, so with `sampling_mode="marginal"` it consumes the whole `q_pi` array at once.

`infer_policies` returns `q_pi` with shape `(batch_size, num_policies)`. `control.get_marginals` (control.py:112) compares it against a condition of shape `(num_controls[f], num_policies)`, and the two broadcast, so the batch axis lands on the action axis whenever numpy lets it: when `batch_size == num_controls[f]`, or when either of them is 1. Otherwise it raises `ValueError`.

One control factor with 3 actions and `policy_len` 1, so the unique multi-actions are actions 0, 1 and 2 in order and the answer is `q_pi` itself:

```python
import jax.numpy as jnp, jax.random as jr
from pymdp import utils
from pymdp.agent import Agent

a_key, b_key = jr.split(jr.PRNGKey(0))
A = utils.random_A_array(a_key, [2], [4], A_dependencies=[[0]])
B = utils.random_B_array(b_key, [4], [3], B_dependencies=[[0]])
q_pi = jnp.array([[0.5, 0.3, 0.2], [0.2, 0.5, 0.3], [0.3, 0.2, 0.5]])

for mode in ("marginal", "full"):
    agent = Agent(A, B, A_dependencies=[[0]], B_dependencies=[[0]],
                  batch_size=3, sampling_mode=mode)
    probs = agent.multiaction_probabilities(q_pi)
    print(mode, probs.shape, float(probs.sum()), probs)
```

```
marginal (3,) 1.5 [0.5 0.5 0.5]
full (3, 3) 3.0 [[0.5 0.3 0.2]
 [0.2 0.5 0.3]
 [0.3 0.2 0.5]]
```

The docstring cuts the other way and I would rather raise that than skip it. agent.py:918 calls `q_pi` "Posterior beliefs over policies for one batch element", which would make the marginal branch the correct one. Its history undercuts it. That wording arrived in `9fa629f`, a sphinx to mkdocs migration, rewriting a docstring that had typed `q_pi` as a 1-D `numpy.ndarray` back when the method carried a method level `@vmap` and the decorator supplied the batch axis. `382ac75` removed the decorator, rewrote both branches into named callables, wrapped only `"full"`, and left the docstring alone. That same migration also rewrote the Returns line to "Probability vector over unique multi-actions", already wrong for the `"full"` branch everything uses.

Everything else reads as batched. `control.get_marginals` documents its own `q_pi` as "1D Array" and `test_sophisticated_inference_jax.py` calls it as `get_marginals(q_pi[0], ...)`, so stripping the batch axis is the caller's job. `sample_action` got "for each batch element" in that same migration and wraps both of its branches in `vmap`. `si_tmaze_SIvalidation.ipynb` passes `qpi_values[:, t, :]` and indexes `[0]`. `docs-mkdocs/migration/numpy-to-jax.md` says most JAX APIs here expect a leading `batch_size` dimension. Under the unbatched reading it is `"full"` and those callers that need changing, and either way the two branches disagree on the same input.

If the batched reading is the intended one, this fixes it for me:

```diff
diff --git a/pymdp/agent.py b/pymdp/agent.py
index 2ca2700..eacb000 100644
--- a/pymdp/agent.py
+++ b/pymdp/agent.py
@@ -925,9 +925,9 @@ class Agent(Module):
 
         if self.sampling_mode == "marginal":
             get_marginals = partial(control.get_marginals, policies=self.policies.policy_arr, num_controls=self.num_controls)
-            marginals = get_marginals(q_pi)
             outer = lambda a, b: jnp.outer(a, b).reshape(-1)
-            marginals = jtu.tree_reduce(outer, marginals)
+            multiaction_probs = lambda x: jtu.tree_reduce(outer, get_marginals(x))
+            marginals = vmap(multiaction_probs)(q_pi)
 
         elif self.sampling_mode == "full":
             locs = jnp.all(
```

The `vmap` has to wrap the `tree_reduce` as well: wrapping only `get_marginals` gives `(24,)` instead of `(2, 6)` for `num_controls=[2, 3]` at `batch_size=2`, since `jnp.outer` then eats the batch axis.

It has a cost. A genuinely 1-D `q_pi` of shape `(num_policies,)`, which is what the docstring asks for, returns `[0.5 0.3 0.2]` in `"marginal"` mode today and `(3, 3)` with this patch. On this one factor example that is the two branches converging rather than new breakage, since `"full"` already returns the same `(3, 3)` for that input on `c19fa9e`. At `(1, num_policies)` the values are unchanged apart from the leading axis.

I have a regression test covering batch sizes 1, 2 and 3 against one 3-action factor. All three fail on `c19fa9e` and pass with the patch. Happy to open a pull request with both if the batched reading is intended. If it is not, the fix belongs in `"full"` and its callers instead, and I would rather hear that first.

Nothing in the repo walks this path today: the one test that calls `multiaction_probabilities` takes the default `sampling_mode="full"`, and `si_tmaze_SIvalidation.ipynb` sets `"full"` explicitly. `"marginal"` is the default in `pymdp/legacy/agent.py`, so it is a plausible thing to carry across when porting.

Not the same as #434, which is about `__init__` and `B_action_dependencies`. This one needs neither.

Ubuntu 24.04.4 LTS, Linux 7.0.0-28-generic, CPython 3.12.3, jax and jaxlib 0.11.0, numpy 2.5.2, CPU only, `XLA_FLAGS=--xla_cpu_experimental_ynn_fusion_type=` as `test.yaml` sets it.
